contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1032,7 +1033,19 @@ update_GrantStmt(Node *n, const char *object, const char *obj_schema, const char
 	if (grantee && stmt->grantees)
 	{
 		RoleSpec   *tmp = (RoleSpec *) llast(stmt->grantees);
-
+		char *cur_dbname = get_cur_db_name();
+		char *pub_grantee = "";
```

Comment:
You have allocated memory and then later you are reassigning the pointer to another address.